packages/flutter/lib/src/material/selectable_text.dart

```diff
@@ -755,51 +755,57 @@ class _SelectableTextState extends State<SelectableText>
-        cursorRadius: cursorRadius,
-        cursorColor: cursorColor,
-        selectionHeightStyle: widget.selectionHeightStyle,
-        selectionWidthStyle: widget.selectionWidthStyle,
-        cursorOpacityAnimates: cursorOpacityAnimates,
-        cursorOffset: cursorOffset,
-        paintCursorAboveText: paintCursorAboveText,
-        backgroundCursorColor: CupertinoColors.inactiveGray,
-        enableInteractiveSelection: widget.enableInteractiveSelection,
-        magnifierConfiguration:
-            widget.magnifierConfiguration ?? TextMagnifier.adaptiveMagnifierConfiguration,
-        dragStartBehavior: widget.dragStartBehavior,
-        scrollPhysics: widget.scrollPhysics,
-        scrollBehavior: widget.scrollBehavior,
-        autofillHints: null,
-        contextMenuBuilder: widget.contextMenuBuilder,
-      ),
+    Widget editable = EditableText(
+      key: editableTextKey,
+      style: effectiveTextStyle,
+      readOnly: true,
+      toolbarOptions: widget.toolbarOptions,
+      textWidthBasis: widget.textWidthBasis ?? defaultTextStyle.textWidthBasis,
+      textHeightBehavior: widget.textHeightBehavior ?? defaultTextStyle.textHeightBehavior,
+      showSelectionHandles: _showSelectionHandles,
+      showCursor: widget.showCursor,
+      controller: _controller,
+      focusNode: focusNode,
+      strutStyle: widget.strutStyle ?? const StrutStyle(),
+      textAlign: widget.textAlign ?? defaultTextStyle.textAlign ?? TextAlign.start,
+      textDirection: widget.textDirection,
+      textScaler: effectiveScaler,
+      autofocus: widget.autofocus,
+      forceLine: false,
+      minLines: widget.minLines,
+      maxLines: widget.maxLines ?? defaultTextStyle.maxLines,
+      selectionColor: widget.selectionColor ?? selectionColor,
+      selectionControls: widget.selectionEnabled ? textSelectionControls : null,
+      onSelectionChanged: _handleSelectionChanged,
+      onSelectionHandleTapped: _handleSelectionHandleTapped,
+      rendererIgnoresPointer: true,
+      cursorWidth: widget.cursorWidth,
+      cursorHeight: widget.cursorHeight,
+      cursorRadius: cursorRadius,
+      cursorColor: cursorColor,
+      selectionHeightStyle: widget.selectionHeightStyle,
+      selectionWidthStyle: widget.selectionWidthStyle,
+      cursorOpacityAnimates: cursorOpacityAnimates,
+      cursorOffset: cursorOffset,
+      paintCursorAboveText: paintCursorAboveText,
+      backgroundCursorColor: CupertinoColors.inactiveGray,
+      enableInteractiveSelection: widget.enableInteractiveSelection,
+      magnifierConfiguration:
+          widget.magnifierConfiguration ?? TextMagnifier.adaptiveMagnifierConfiguration,
+      dragStartBehavior: widget.dragStartBehavior,
+      scrollPhysics: widget.scrollPhysics,
+      scrollBehavior: widget.scrollBehavior,
+      autofillHints: null,
+      contextMenuBuilder: widget.contextMenuBuilder,
     );
+    final ui.TypographySettings? typographySettings = MediaQuery.maybeTypographySettingsOf(context);
```

Comment:
Not something that blocks this pr

I noticed we start to put more and more text related settings into MediaQuery, but we also have DefaultTextStyle. What are the guideline to decide whether something should be in one vs the other?

Reply:
In this case it feels like it would still be important to distinguish between application level configuration and system wide configuration in the case one is preferred over the other. It also feels like this type of system setting is unlikely to exposed because it is a setting that would affect many other apps as well, so the overlap might not be something we have to consider.

Reply:
MediaQuery should be considered application level configuration. It is just by default Material app will inherited system level configuration into default MediaQuery. The developer can still override them if they want.

I guess my question is why don't we put TextStyle direclty in MediaQuery to carry the things like text space, or let TextStyle carry a TypographySettings as its property and remove the text spacing, line high...etc.

Reply:
yup that sounds good to me.